draft-ietf-ipsecme-ikev2-multiple-ke-00.xml

```diff
@@ -0,0 +1,1037 @@
+        </t>
+
+        </list>
+    	</t>
+
+	</section>
+
+	<section title="The Framework of Hybrid Post-Quantum Key Exchange" anchor="framework">
+		<section title="Overall design">
+            <t> This design assigns new Transform Type 4 identifiers to the
+			various post-quantum key exchanges (which will be defined later).  We
+			specifically do not make a distinction between classical (DH and ECDH)
+			and post-quantum key exchanges, nor post-quantum algorithms which are
+			true key exchanges versus post-quantum algorithms that act as key
+			transport mechanisms; all are treated equivalently by the
+			protocol. To make this more clear for implementers this document
+            renames Transform Type 4 from "Diffie-Hellman Group (D-H)"
+            to "Key Exchange Method (KE)" and renames a field in the Key Exchange Payload from "Diffie-Hellman Group Num"
+            to "Key Exchange Method". The corresponding IANA registry is also renamed
+            from "Diffie-Hellman Group Transform IDs" to "Key Exchange Method Transform IDs".
+            </t>
+
+            <t> In order to support IKE fragmentation for additional key exchanges
+            that may have a long public keys, the proposed framework utilizes the IKE_INTERMEDIATE exchange 
+            defined in <xref target="I-D.ietf-ipsecme-ikev2-intermediate"/>.
+            </t>
+
+            <t> In order to minimize communication overhead, only the key shares that are agreed to be used
+			are actually exchanged. In order to achieve this several new
+            Transform Types are defined, each sharing possible Transform IDs 
+            with Transform Type 4. The IKE_SA_INIT message includes one or more newly defined SA transforms 
+			that lists the extra key exchange policy required by the
+			initiator; the responder selects single transform of each type, and
+			returns them back in the response IKE_SA_INIT message. 
+            Then, provided that additional key exchanges are negotiated the initiator and the responder 
+            perform one or more IKE_INTERMEDIATE exchanges; each such exchange includes a KE payload 
+            for one of the negotiated key exchanges.</t>
+
+            <t>Here is an overview of the initial exchanges:</t>
+
+            <figure><artwork align="center" ><![CDATA[
+Initiator                             Responder
+---------------------------------------------------------------------
+<-- IKE_SA_INIT (additional key exchanges negotiation) -->
+
+<-- {IKE_INTERMEDIATE (additional key exchange)} -->
+
+                         ...
+
+<-- {IKE_INTERMEDIATE (additional key exchange)} -->
+
+<-- {IKE_AUTH} -->
+            ]]></artwork></figure>
+
+            <t> The additional key exchanges can use algorithms that are
+    		currently considered to be resistant to quantum computer attacks.  These
+    		algorithms are collectively referred to as post-quantum algorithms in
+    		this document.</t>
```

Comment:
Perhaps we should add the following sentence to emphasise that classical primitives could be used too:

```suggestion
    		this document. However, it is also possible to use classical (EC)DH primitives for non post-quantum requirements.</t>
```